fiks-meldingsformidler/src/main/java/no/difi/meldingsutveksling/ks/svarut/SvarUtService.java

```diff
@@ -1,61 +1,74 @@
+import no.difi.meldingsutveksling.nextmove.NextMoveException;
+import no.difi.meldingsutveksling.nextmove.NextMoveOutMessage;
 import no.difi.meldingsutveksling.receipt.Conversation;
-import no.difi.meldingsutveksling.receipt.GenericReceiptStatus;
 import no.difi.meldingsutveksling.receipt.MessageStatus;
+import no.difi.meldingsutveksling.serviceregistry.SRParameter;
 import no.difi.meldingsutveksling.serviceregistry.ServiceRegistryLookup;
+import no.difi.meldingsutveksling.serviceregistry.ServiceRegistryLookupException;
 import no.difi.meldingsutveksling.serviceregistry.externalmodel.ServiceRecord;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
 
 import java.security.cert.X509Certificate;
-import java.time.LocalDateTime;
-import java.util.Optional;
 
+@Component
+@ConditionalOnProperty(name = "difi.move.feature.enableDPF", havingValue = "true")
+@RequiredArgsConstructor
 public class SvarUtService {
-    private SvarUtWebServiceClient client;
-    private ServiceRegistryLookup serviceRegistryLookup;
-    private FiksMapper fiksMapper;
-    private IntegrasjonspunktProperties props;
-    CertificateParser certificateParser;
 
-    public SvarUtService(SvarUtWebServiceClient svarUtClient,
-                         ServiceRegistryLookup serviceRegistryLookup,
-                         FiksMapper fiksMapper,
-                         IntegrasjonspunktProperties props) {
-        this.client = svarUtClient;
-        this.serviceRegistryLookup = serviceRegistryLookup;
-        this.fiksMapper = fiksMapper;
-        this.props = props;
-        certificateParser = new CertificateParser();
-    }
+    private final SvarUtWebServiceClient client;
+    private final ServiceRegistryLookup serviceRegistryLookup;
+    private final FiksMapper fiksMapper;
+    private final IntegrasjonspunktProperties props;
+    private final CertificateParser certificateParser;
+    private final FiksStatusMapper fiksStatusMapper;
 
-    public String send(EDUCore message) {
-        Optional<ServiceRecord> serviceRecord = serviceRegistryLookup.getServiceRecord(message.getReceiver().getIdentifier(), message.getServiceIdentifier());
-        if (!serviceRecord.isPresent()) {
-            throw new SvarUtServiceException(String.format("No DPF ServiceRecord found for identifier %s", message.getReceiver().getIdentifier()));
+    public String send(NextMoveOutMessage message) throws NextMoveException {
+        ServiceRecord serviceRecord;
+        try {
+            serviceRecord = serviceRegistryLookup.getServiceRecord(SRParameter.builder(message.getReceiverIdentifier())
+                            .securityLevel(message.getBusinessMessage().getSikkerhetsnivaa())
+                            .conversationId(message.getConversationId()).build(),
+                    message.getServiceIdentifier());
+        } catch (ServiceRegistryLookupException e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1166)